packages/commerce/frontend/src/components/header.tsx

```diff
@@ -9,14 +9,18 @@ export const Header = () => {
         <MainNav />
         <div className="flex items-center justify-end">
           <nav className="flex items-center">
-            <NavLink to="/cart">
-              <ShoppingCart className="h-6 w-6 text-green-3 transition-colors hover:text-green" />
-              <span className="sr-only">Shopping cart</span>
-            </NavLink>
+            <ShoppingCartPopover />
+            <ShoppingCartTotalItems />
           </nav>
           <MobileNav />
         </div>
       </div>
     </header>
   )
 }
+
+const ShoppingCartTotalItems = () => {
+  const { totalItems } = useCartStore()
+
+  return <span className="text-sm">{totalItems}</span>
+}
```

Comment:
Moved this to a separate component to avoid header re-renders when adding items to the cart.